src/cpp/cpp_typecheck_conversions.cpp

```diff
@@ -1333,8 +1333,9 @@ bool cpp_typecheckt::reference_binding(
            reference_compatible(returned_value, type, rank))
         {
           // returned values are lvalues in case of references only
-          assert(returned_value.id()==ID_dereference &&
-                 is_reference(returned_value.op0().type()));
+          assert(
+            returned_value.id() == ID_dereference &&
```

Comment:
Seems unnecessary as the second conjunct now uses `to_dereference_expr`.

Reply:
removed